### PR TITLE
Do not consume `self` on `digest`

### PR DESCRIPTION
**Pull request by D4nte**
Saves having a clone every time `digest` is called.

### CODE REVIEW

File: cnd/src/network/protocols/announce.rs [hunk 1]
```diff
@@ -17,8 +17,8 @@ impl SwapDigest {
 }
 
 impl IntoDigestInput for SwapDigest {
-    fn into_digest_input(self) -> Vec<u8> {
-        self.0.into_bytes()
+    fn into_digest_input(&self) -> Vec<u8> {
```
Comment (thomaseizinger) [inline cnd/src/network/protocols/announce.rs:20]: Should be called `to_digest_input` according to Rust conventions :)

File: digest/src/lib.rs [hunk 2]
```diff
@@ -67,25 +67,25 @@ pub use digest_macro_derive::Digest;
 pub use hex;
 
 pub trait Digest {
-    type Hash: Hash + IntoDigestInput;
-    fn digest(self) -> Self::Hash;
+    type Hash: Hash + ToDigestInput;
```
Comment (thomaseizinger) [inline digest/src/lib.rs:70]: I actually didn't notice before, why are we requiring this trait bound here?

Comment (D4nte) [reply digest/src/lib.rs:70]: Can you read the doc in the `lib.rs` and let me know if the question still holds?

Comment (thomaseizinger) [reply digest/src/lib.rs:70]: Makes sense now, thanks :)
I think I was mixing it up with the `Digest` trait somehow.

**Pull request merged**